From d98ef3e18254b1d05064f6ab459970efec9d8031 Mon Sep 17 00:00:00 2001
From: Krzysztof Kolasa <kkolasa@winsoft.pl>
Date: Tue, 10 Sep 2013 21:45:50 +0200
Subject: [PATCH] Changed module license string on "GPL" for kernel
 next-20130910, future 3.12.0-rc1

---
 firegl_public.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/firegl_public.c b/firegl_public.c
index 9b796bc..ff87ce4 100644
--- a/firegl_public.c
+++ b/firegl_public.c
@@ -259,7 +259,13 @@
 #endif
 
 #ifdef MODULE_LICENSE
+
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(3,11,0)
 MODULE_LICENSE("Proprietary. (C) 2002 - ATI Technologies, Starnberg, GERMANY");
+#else
+MODULE_LICENSE("GPL");
+#endif
+
 #endif
 #ifdef MODULE_DEVICE_TABLE
 MODULE_DEVICE_TABLE(pci, fglrx_pci_table);
-- 
1.8.4

