--- 13.8/common/lib/modules/fglrx/build_mod/firegl_public.c 2013-08-19 23:10:07.000000000 +0200
+++ 13.8/common/lib/modules/fglrx/build_mod/firegl_public.c 2013-08-19 23:10:07.000000000 +0200
@@ -821,7 +821,7 @@
     KCL_DEBUG1(FN_FIREGL_PROC, "minor %d, proc_list 0x%08lx\n", minor, (unsigned long)proc_list);
     if (!minor)
     {
-        root = KCL_create_proc_dir(NULL, "ati", S_IFDIR);
+        root = KCL_create_proc_dir(NULL, "ati", S_IFDIR|S_IRUGO|S_IXUGO);
     }
 
     if (!root)
@@ -847,7 +847,7 @@ static struct proc_dir_entry *firegl_proc_init( device_t *dev,
     }
 
     sprintf(name, "%d", minor);
-    *dev_root = KCL_create_proc_dir(root, name, S_IFDIR);
+    *dev_root = KCL_create_proc_dir(root, name, S_IRUGO|S_IXUGO);
     if (!*dev_root) {
         KCL_remove_proc_dir_entry(root, "major");
         KCL_remove_proc_dir_entry(NULL, "ati");
--- 13.8/common/lib/modules/fglrx/build_mod/drm_proc.h 2013-08-19 23:10:07.000000000 +0200
+++ 13.8/common/lib/modules/fglrx/build_mod/drm_proc.h 2013-08-19 23:10:07.000000000 +0200
@@ -97,7 +97,7 @@ struct proc_dir_entry *DRM(proc_init)(drm_device_t *dev, int minor,
 
         if (!minor)
         {
-            root = KCL_create_proc_dir(NULL, "dri", S_IFDIR);
+            root = KCL_create_proc_dir(NULL, "dri", S_IRUGO|S_IXUGO);
         }
         if (!root) 
         {
@@ -106,7 +106,7 @@ struct proc_dir_entry *DRM(proc_init)(drm_device_t *dev, int minor,
         }
 
         sprintf(name, "%d", minor);
-        *dev_root = KCL_create_proc_dir(root, name, S_IFDIR);
+        *dev_root = KCL_create_proc_dir(root, name, S_IRUGO|S_IXUGO);
         if (!*dev_root) 
         {
             DRM_ERROR("Cannot create /proc/ati/%s\n", name);
