From 5f92d49637f73101d62eb8180a24d8d13b5205c4 Mon Sep 17 00:00:00 2001
From: Krzysztof Kolasa <kkolasa@winsoft.pl>
Date: Tue, 17 Sep 2013 18:54:55 +0200
Subject: [PATCH] Add string "GPL\0" for kernel 3.12.0-rc1

---
 firegl_public.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/firegl_public.c b/firegl_public.c
index ff87ce4..4e1a9d5 100644
--- a/firegl_public.c
+++ b/firegl_public.c
@@ -260,10 +260,10 @@
 
 #ifdef MODULE_LICENSE
 
-#if LINUX_VERSION_CODE <= KERNEL_VERSION(3,11,0)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,12,0)
 MODULE_LICENSE("Proprietary. (C) 2002 - ATI Technologies, Starnberg, GERMANY");
 #else
-MODULE_LICENSE("GPL");
+MODULE_LICENSE("GPL\0Proprietary. (C) 2002 - ATI Technologies, Starnberg, GERMANY");
 #endif
 
 #endif
-- 
1.8.4

