*** src/lxdm.c.orig	2011-07-29 15:42:33.000000000 +0200
--- src/lxdm.c	2012-09-20 18:55:05.375316290 +0200
***************
*** 1202,1207 ****
--- 1202,1214 ----
  	}
  	gchar *argv[] = { "/etc/lxdm/PostLogout", NULL };
  	g_spawn_async(NULL, argv, s->env, G_SPAWN_SEARCH_PATH, NULL, NULL, NULL, NULL);
+ 
+ 	// Restart X server if set in config
+ 	if (g_key_file_get_integer(config,"server","restart_after_logout",NULL) == 1)
+ 	{
+         lxsession_free(s);
+         lxsession_greeter();
+ 	}
  }
  
  gboolean lxdm_get_session_info(char *session,char **pname,char **pexec)
