diff -Npur xfce4-pulseaudio-plugin-0.2.4.orig/panel-plugin/pulseaudio-button.c xfce4-pulseaudio-plugin-0.2.4/panel-plugin/pulseaudio-button.c
--- xfce4-pulseaudio-plugin-0.2.4.orig/panel-plugin/pulseaudio-button.c	2015-06-06 23:57:55.000000000 +0200
+++ xfce4-pulseaudio-plugin-0.2.4/panel-plugin/pulseaudio-button.c	2015-12-13 12:30:28.222054714 +0100
@@ -53,10 +53,10 @@
 
 /* Icons for different volume levels */
 static const char *icons[] = {
-  "audio-volume-muted-symbolic",
-  "audio-volume-low-symbolic",
-  "audio-volume-medium-symbolic",
-  "audio-volume-high-symbolic",
+  "audio-volume-muted-panel",
+  "audio-volume-low-panel",
+  "audio-volume-medium-panel",
+  "audio-volume-high-panel",
   NULL
 };
 
diff -Npur xfce4-pulseaudio-plugin-0.2.4.orig/panel-plugin/pulseaudio-notify.c xfce4-pulseaudio-plugin-0.2.4/panel-plugin/pulseaudio-notify.c
--- xfce4-pulseaudio-plugin-0.2.4.orig/panel-plugin/pulseaudio-notify.c	2015-06-10 23:56:50.000000000 +0200
+++ xfce4-pulseaudio-plugin-0.2.4/panel-plugin/pulseaudio-notify.c	2015-12-13 12:30:28.228724714 +0100
@@ -51,10 +51,10 @@
 
 /* Icons for different volume levels */
 static const char *icons[] = {
-  "audio-volume-muted-symbolic",
-  "audio-volume-low-symbolic",
-  "audio-volume-medium-symbolic",
-  "audio-volume-high-symbolic",
+  "audio-volume-muted-panel",
+  "audio-volume-low-panel",
+  "audio-volume-medium-panel",
+  "audio-volume-high-panel",
   NULL
 };
 
