diff -Npur NVIDIA-Linux-x86_64-304.131-no-compat32.orig/kernel/nv-mlock.c NVIDIA-Linux-x86_64-304.131-no-compat32/kernel/nv-mlock.c
--- NVIDIA-Linux-x86_64-304.131-no-compat32.orig/kernel/nv-mlock.c	2016-10-23 12:46:29.567949000 +0200
+++ NVIDIA-Linux-x86_64-304.131-no-compat32/kernel/nv-mlock.c	2016-10-23 12:49:39.888275125 +0200
@@ -51,7 +51,7 @@ RM_STATUS NV_API_CALL nv_lock_user_pages
 
     down_read(&mm->mmap_sem);
     ret = NV_GET_USER_PAGES((unsigned long)address,
-            page_count, write, force, user_pages, NULL);
+            page_count, write ? FOLL_WRITE : 0, user_pages, NULL);
     up_read(&mm->mmap_sem);
     pinned = ret;
 
diff -Npur NVIDIA-Linux-x86_64-304.131-no-compat32.orig/kernel/nv-mm.h NVIDIA-Linux-x86_64-304.131-no-compat32/kernel/nv-mm.h
--- NVIDIA-Linux-x86_64-304.131-no-compat32.orig/kernel/nv-mm.h	2016-10-23 12:46:29.567949000 +0200
+++ NVIDIA-Linux-x86_64-304.131-no-compat32/kernel/nv-mm.h	2016-10-23 12:49:15.166938027 +0200
@@ -45,8 +45,8 @@
     #define NV_GET_USER_PAGES           get_user_pages
     #define NV_GET_USER_PAGES_REMOTE    get_user_pages_remote
 #else
-    #define NV_GET_USER_PAGES(start, nr_pages, write, force, pages, vmas) \
-        get_user_pages(current, current->mm, start, nr_pages, write, force, pages, vmas)
+    #define NV_GET_USER_PAGES(start, nr_pages, gup_flags, pages, vmas) \
+        get_user_pages(current, current->mm, start, nr_pages, gup_flags, pages, vmas)
 
     #define NV_GET_USER_PAGES_REMOTE    get_user_pages
 #endif
