diff -rupN a/os/linux/rt_linux.c b/os/linux/rt_linux.c
--- a/os/linux/rt_linux.c	2016-07-17 22:34:01.000000000 +0200
+++ b/os/linux/rt_linux.c	2017-01-15 18:43:05.551892618 +0100
@@ -1201,13 +1201,21 @@ void RtmpOSFileSeek(RTMP_OS_FD osfd, int
 int RtmpOSFileRead(RTMP_OS_FD osfd, char *pDataPtr, int readLen)
 {
 	// The object must have a read method
-	return __vfs_read(osfd, pDataPtr, readLen, &osfd->f_pos);
+	if (osfd->f_op && osfd->f_op->read)
+	{
+		return osfd->f_op->read(osfd,  pDataPtr, readLen, &osfd->f_pos);
+	}
+	else
+	{
+		DBGPRINT(RT_DEBUG_ERROR, ("no file read method\n"));
+		return -1;
+	}
 }
 
 
 int RtmpOSFileWrite(RTMP_OS_FD osfd, char *pDataPtr, int writeLen)
 {
-    return __vfs_write(osfd, pDataPtr, (size_t) writeLen, &osfd->f_pos);
+	return osfd->f_op->write(osfd, pDataPtr, (size_t)writeLen, &osfd->f_pos);
 }
 
 
