--- a/kernel/nv.c	2017-11-25 07:14:29.000000000 -0600
+++ b/kernel/nv.c	2017-11-27 00:09:01.948814571 -0600
@@ -332,7 +332,7 @@
 #else
 static irqreturn_t   nvidia_isr            (int, void *);
 #endif
-static void          nvidia_rc_timer       (unsigned long);
+static void          nvidia_rc_timer       (struct timer_list *t);
 
 static int           nvidia_ctl_open       (struct inode *, struct file *);
 static int           nvidia_ctl_close      (struct inode *, struct file *);
@@ -2472,10 +2472,10 @@
 
 static void
 nvidia_rc_timer(
-    unsigned long data
+    struct timer_list *t 
 )
 {
-    nv_linux_state_t *nvl = (nv_linux_state_t *) data;
+    nv_linux_state_t *nvl = from_timer(nvl, t, rc_timer);
     nv_state_t *nv = NV_STATE_PTR(nvl);
     nvidia_stack_t *sp = nvl->sp[NV_DEV_STACK_TIMER];
 
@@ -3386,9 +3386,7 @@
         return -1;
 
     nv_printf(NV_DBG_INFO, "NVRM: initializing rc timer\n");
-    init_timer(&nvl->rc_timer);
-    nvl->rc_timer.function = nvidia_rc_timer;
-    nvl->rc_timer.data = (unsigned long) nvl;
+    timer_setup(&nvl->rc_timer, nvidia_rc_timer, 0);
     nv->rc_timer_enabled = 1;
     mod_timer(&nvl->rc_timer, jiffies + HZ); /* set our timeout for 1 second */
     nv_printf(NV_DBG_INFO, "NVRM: rc timer initialized\n");
