diff -ru NVIDIA-Linux-x86_64-375.39/kernel/common/inc/nv-mm.h NVIDIA-Linux-x86_64-375.39-r1/kernel/common/inc/nv-mm.h
--- NVIDIA-Linux-x86_64-375.39/kernel/common/inc/nv-mm.h	2017-01-31 21:50:37.000000000 -0500
+++ NVIDIA-Linux-x86_64-375.39-r1/kernel/common/inc/nv-mm.h	2017-03-30 23:36:39.392964700 -0400
@@ -131,11 +131,7 @@
 #if defined(NV_VM_FAULT_PRESENT)
     static inline unsigned long nv_page_fault_va(struct vm_fault *vmf)
     {
-    #if defined(NV_VM_FAULT_HAS_ADDRESS)
         return vmf->address;
-    #else
-        return (unsigned long)(uintptr_t)(vmf->virtual_address);
-    #endif
     }
 #endif // NV_VM_FAULT_PRESENT
 
diff -ru NVIDIA-Linux-x86_64-375.39/kernel/nvidia-drm/nvidia-drm-drv.c NVIDIA-Linux-x86_64-375.39-r1/kernel/nvidia-drm/nvidia-drm-drv.c
--- NVIDIA-Linux-x86_64-375.39/kernel/nvidia-drm/nvidia-drm-drv.c	2017-01-31 21:47:52.000000000 -0500
+++ NVIDIA-Linux-x86_64-375.39-r1/kernel/nvidia-drm/nvidia-drm-drv.c	2017-03-30 23:45:26.507958893 -0400
@@ -371,7 +371,7 @@
     return 0;
 }
 
-static int nvidia_drm_unload(struct drm_device *dev)
+static void nvidia_drm_unload(struct drm_device *dev)
 {
 #if defined(NV_DRM_ATOMIC_MODESET_AVAILABLE)
     struct NvKmsKapiDevice *pDevice = NULL;
@@ -385,7 +385,7 @@
 
     if (!nvidia_drm_modeset_enabled(dev))
     {
-        return 0;
+        return;
     }
 
     mutex_lock(&nv_dev->lock);
@@ -418,7 +418,7 @@
 
 #endif /* NV_DRM_ATOMIC_MODESET_AVAILABLE */
 
-    return 0;
+    return;
 }
 
 #if defined(NV_DRM_ATOMIC_MODESET_AVAILABLE)
diff -ru NVIDIA-Linux-x86_64-375.39/kernel/nvidia-drm/nvidia-drm-encoder.h NVIDIA-Linux-x86_64-375.39-r1/kernel/nvidia-drm/nvidia-drm-encoder.h
--- NVIDIA-Linux-x86_64-375.39/kernel/nvidia-drm/nvidia-drm-encoder.h	2017-01-31 21:47:52.000000000 -0500
+++ NVIDIA-Linux-x86_64-375.39-r1/kernel/nvidia-drm/nvidia-drm-encoder.h	2017-03-30 17:19:04.373899579 -0400
@@ -32,6 +32,8 @@
 #include <drm/drmP.h>
 #include "nvkms-kapi.h"
 
+#include <drm/drm_encoder.h>
+
 struct nvidia_drm_encoder
 {
     NvKmsKapiDisplay hDisplay;
diff -ru NVIDIA-Linux-x86_64-375.39/kernel/nvidia-drm/nvidia-drm-fb.c NVIDIA-Linux-x86_64-375.39-r1/kernel/nvidia-drm/nvidia-drm-fb.c
--- NVIDIA-Linux-x86_64-375.39/kernel/nvidia-drm/nvidia-drm-fb.c	2017-01-31 21:47:52.000000000 -0500
+++ NVIDIA-Linux-x86_64-375.39-r1/kernel/nvidia-drm/nvidia-drm-fb.c	2017-03-30 23:40:17.097962302 -0400
@@ -77,7 +77,7 @@
 static struct drm_framebuffer *internal_framebuffer_create
 (
     struct drm_device *dev,
-    struct drm_file *file, NV_DRM_MODE_FB_CMD2_T *cmd,
+    struct drm_file *file, const struct drm_mode_fb_cmd2 *cmd,
     uint64_t nvkms_params_ptr,
     uint64_t nvkms_params_size
 )
@@ -153,7 +153,7 @@
 
     /* Fill out framebuffer metadata from the userspace fb creation request */
 
-    drm_helper_mode_fill_fb_struct(&nv_fb->base, cmd);
+    drm_helper_mode_fill_fb_struct(dev, &nv_fb->base, cmd);
 
     /* Initialize the base framebuffer object and add it to drm subsystem */
 
@@ -199,7 +199,7 @@
 struct drm_framebuffer *nvidia_drm_framebuffer_create
 (
     struct drm_device *dev,
-    struct drm_file *file, NV_DRM_MODE_FB_CMD2_T *cmd
+    struct drm_file *file, const struct drm_mode_fb_cmd2 *cmd
 )
 {
     return internal_framebuffer_create(dev, file, cmd, 0, 0);
diff -ru NVIDIA-Linux-x86_64-375.39/kernel/nvidia-drm/nvidia-drm-fb.h NVIDIA-Linux-x86_64-375.39-r1/kernel/nvidia-drm/nvidia-drm-fb.h
--- NVIDIA-Linux-x86_64-375.39/kernel/nvidia-drm/nvidia-drm-fb.h	2017-01-31 21:47:52.000000000 -0500
+++ NVIDIA-Linux-x86_64-375.39-r1/kernel/nvidia-drm/nvidia-drm-fb.h	2017-03-30 23:30:29.119968779 -0400
@@ -45,7 +45,7 @@
 struct drm_framebuffer *nvidia_drm_framebuffer_create
 (
     struct drm_device *dev,
-    struct drm_file *file, NV_DRM_MODE_FB_CMD2_T *cmd
+    struct drm_file *file, const struct drm_mode_fb_cmd2 *cmd
 );
 
 int nvidia_drm_add_nvkms_fb(
diff -ur NVIDIA-Linux-x86_64-375.39/kernel/nvidia-drm/nvidia-drm-linux.c NVIDIA-Linux-x86_64-381.09.patched/kernel/nvidia-drm/nvidia-drm-linux.c
--- NVIDIA-Linux-x86_64-375.39/kernel/nvidia-drm/nvidia-drm-linux.c	2017-03-31 03:42:21.000000000 +0200
+++ NVIDIA-Linux-x86_64-375.39.patched/kernel/nvidia-drm/nvidia-drm-linux.c	2017-04-06 23:53:14.273356795 +0200
@@ -185,7 +185,7 @@
 module_exit(nv_linux_drm_exit);
 
 #if defined(MODULE_LICENSE)
-  MODULE_LICENSE("MIT");
+  MODULE_LICENSE("Dual MIT/GPL");
 #endif
 #if defined(MODULE_INFO)
   MODULE_INFO(supported, "external");
--- NVIDIA-Linux-x86_64-375.39/kernel/nvidia-modeset/nvidia-modeset-linux.c	2017-02-01 03:47:52.000000000 +0100
+++ NVIDIA-Linux-x86_64-375.39.patched/kernel/nvidia-modeset/nvidia-modeset-linux.c.n	2017-05-02 11:24:07.240958498 +0200
@@ -1285,7 +1285,7 @@ module_init(nvkms_init);
 module_exit(nvkms_exit);
 
 #if defined(MODULE_LICENSE)
-  MODULE_LICENSE("NVIDIA");
+  MODULE_LICENSE("GPL\0NVIDIA");
 #endif
 #if defined(MODULE_INFO)
   MODULE_INFO(supported, "external");
