diff -ur NVIDIA-Linux-x86_64-375.39/kernel/nvidia-drm/nvidia-drm-linux.c NVIDIA-Linux-x86_64-381.09.patched/kernel/nvidia-drm/nvidia-drm-linux.c
--- NVIDIA-Linux-x86_64-375.39/kernel/nvidia-drm/nvidia-drm-linux.c	2017-03-31 03:42:21.000000000 +0200
+++ NVIDIA-Linux-x86_64-375.39.patched/kernel/nvidia-drm/nvidia-drm-linux.c	2017-04-06 23:53:14.273356795 +0200
@@ -185,7 +185,7 @@
 module_exit(nv_linux_drm_exit);
 
 #if defined(MODULE_LICENSE)
-  MODULE_LICENSE("MIT");
+  MODULE_LICENSE("Dual MIT/GPL");
 #endif
 #if defined(MODULE_INFO)
   MODULE_INFO(supported, "external");
