diff -rupN a/dkms b/dkms
--- a/dkms	2017-06-21 03:16:09.318322820 +0200
+++ b/dkms	2017-06-21 03:15:23.004366460 +0200
@@ -1522,12 +1522,6 @@ install_module()
             exit 6
     }
 
-    # Make the newly installed modules available immediately
-    find /sys/devices -name modalias -print0 | xargs -0 cat | xargs modprobe -a -b -q
-    if [ -f /lib/systemd/system/systemd-modules-load.service ]; then
-        systemctl restart systemd-modules-load.service
-    fi
-
     # Do remake_initrd things (save old initrd)
     [[ $remake_initrd ]] && ! make_initrd "$kernelver" "$arch" && {
         do_uninstall "$kernelver" "$arch"
