--- a/driver/usb.c	2016-10-09 11:45:08.000000000 -0400
+++ b/driver/usb.c	2017-11-26 08:58:24.499159544 -0500
@@ -776,8 +776,8 @@ static USBD_STATUS wrap_get_status_reque
 		return NT_URB_STATUS(nt_urb);
 	}
 	assert(status_req->transfer_buffer_length == sizeof(u16));
-	ret = usb_get_status(udev, type, status_req->index,
-			     status_req->transfer_buffer);
+	ret = usb_get_std_status(udev, type, status_req->index,
+			         status_req->transfer_buffer);
 	if (ret >= 0) {
 		assert(ret <= status_req->transfer_buffer_length);
 		status_req->transfer_buffer_length = ret;
