Index: trunk/src/VBox/Runtime/r0drv/linux/threadctxhooks-r0drv-linux.c
===================================================================
--- a/vboxdrv/r0drv/linux/threadctxhooks-r0drv-linux.c
+++ b/vboxdrv/r0drv/linux/threadctxhooks-r0drv-linux.c
@@ -191,4 +191,8 @@
     pThis->PreemptOps.sched_in  = rtThreadCtxHooksLnxSchedIn;
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 2, 0)
+    preempt_notifier_inc();
+#endif
+
     *phCtxHook = pThis;
     return VINF_SUCCESS;
@@ -220,4 +224,8 @@
         Assert(!pThis->fEnabled); /* paranoia */
     }
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 2, 0)
+    preempt_notifier_dec();
+#endif
 
     ASMAtomicWriteU32(&pThis->u32Magic, ~RTTHREADCTXHOOKINT_MAGIC);

